airflow/providers/amazon/aws/operators/eks.py

```diff
@@ -1100,4 +1108,22 @@ def execute(self, context: Context):
         with eks_hook.generate_config_file(
             eks_cluster_name=self.cluster_name, pod_namespace=self.namespace
         ) as self.config_file:
+            if self.deferrable:
```

Comment:
You are waiting before actually executing the task? Also, you need to check `wait_for_completion`, if `wait_for_completion` is `False` you do not want to wait. `wait_for_completion` configures whether the operator wait for a given status, `deferrable` configures how to wait (if `wait_for_completion` is `True`): in poke mode or deferrable mode.

Reply:
Thank you for review. I'm still getting familiar with deferrable concept and will work on resolving the issue as I understand it better.
I'll proceed accordingly, and mention if I need any help!

Reply:
Perfect! Thanks for doing it :)

Reply:
@vincbeck i need help! 😊
Could i need to add unit test in this PR?
in other operator test code amazon provider, i don't find defer test.
Are there any reference unit-test codes?